src/main/java/org/folio/config/ConfigurationsClient.java

```diff
@@ -139,96 +88,50 @@ protected static void verifyOkapiHeaders(OkapiHeaders okapiHeaders) throws Missi
       .put("value", value);
 
     // decide to POST or PUT
-    checkEntry(okapiHeaders, code).onComplete(checkHandler -> {
-      if (checkHandler.failed()) {
-        result.fail(checkHandler.cause());
-      } else {
-        String configId = checkHandler.result();
-
+    return checkEntry(vertx, okapiHeaders, code)
+      .compose(configId -> {
         // not existing -> POST, existing->PUT
         HttpMethod httpMethod = configId == null ? HttpMethod.POST : HttpMethod.PUT;
         String endpoint = configId == null ? CONFIGURATIONS_ENTRIES_ENDPOINT_URL : CONFIGURATIONS_ENTRIES_ENDPOINT_URL + "/" + configId;
 
-        Map<String, String> headers = new HashMap<>();
-        headers.put(XOkapiHeaders.TOKEN, okapiHeaders.getToken());
-
-        try {
-          HttpClientInterface storeEntryClient = HttpClientFactory.getHttpClient(okapiHeaders.getUrl(), okapiHeaders.getTenant(), true);
-          storeEntryClient.setDefaultHeaders(headers);
-          storeEntryClient.request(httpMethod, requestBody, endpoint, null)
-            .whenComplete((storeEntryResponse, throwable) -> {
-
-              if (storeEntryResponse == null) {
-                if (throwable == null) {
-                  result.fail("Cannot " + httpMethod.toString() + " configuration entry");
-                } else {
-                  result.fail(throwable);
-                }
-              }
-              // POST->201 created, PUT->204 no content
-              else if ((httpMethod.equals(HttpMethod.POST) && storeEntryResponse.getCode() == 201)
-                || (httpMethod.equals(HttpMethod.PUT) && storeEntryResponse.getCode() == 204)) {
-
-                result.complete();
-              } else {
-                result.fail("The response status is not 'created',instead "
-                  + storeEntryResponse.getCode()
-                  + " with message  "
-                  + storeEntryResponse.getError());
-              }
-
-            });
-        } catch (Exception ex) {
-          result.fail(ex);
-        }
-      }
-    });
-
-
-    return result.future();
+        return WebClientFactory.getWebClient(vertx)
+          .requestAbs(httpMethod, okapiHeaders.getUrl() + endpoint)
+          .putHeader(XOkapiHeaders.TOKEN, okapiHeaders.getToken())
+          .putHeader(XOkapiHeaders.URL, okapiHeaders.getUrl())
+          .putHeader(XOkapiHeaders.TENANT, okapiHeaders.getTenant())
+          .sendJsonObject(requestBody)
+          .map(res -> {
+            if (res.statusCode() != 200 && res.statusCode() != 201 && res.statusCode() != 204) {
+              throw new RuntimeException("Could not store entry code=" + code + " value=" + value + " status= " + res.statusCode());
+            }
+            return null;
+          });
```

Comment:
```suggestion
          .expect(ResponsePredicate.status(201, 204))
          .sendJsonObject(requestBody)
          .mapEmpty();
```
https://vertx.io/docs/vertx-web-client/java/#_predefined_predicates
Neither PUT nor POST return 200 on success.

Reply:
OK. It's actually 201, 205. Weird..